### PR TITLE
[Mellanox] Add python3 support for Mellanox platform API

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
**- Why I did it**

python2 is end of life and SONiC is going to support python3. This PR is going to support:

1. Mellanox SONiC platform API python3 support
2. Install both python2 and python3 verson of Mellanox SONiC platform API or pmon and host side

**- How I did it**

1. Change docker-platform-monitor make files to install both python2 and python3 whl
2. Change Mellanox SONiC platform API code to support both python2 and python3

**- How to verify it**

Run platform regressions on a few platforms

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: files/build_templates/sonic_debian_extension.j2 [hunk 1]
```diff
@@ -643,6 +643,12 @@ MLNX_PLATFORM_COMMON_PY2_WHEEL_NAME=$(basename {{mlnx_platform_api_py2_wheel_pat
 sudo cp {{mlnx_platform_api_py2_wheel_path}} $FILESYSTEM_ROOT/$MLNX_PLATFORM_COMMON_PY2_WHEEL_NAME
 sudo https_proxy=$https_proxy LANG=C chroot $FILESYSTEM_ROOT pip2 install $MLNX_PLATFORM_COMMON_PY2_WHEEL_NAME
 sudo rm -rf $FILESYSTEM_ROOT/$MLNX_PLATFORM_COMMON_PY2_WHEEL_NAME
+
+# Install mlnx-sonic-platform-common Python 3 package
+MLNX_PLATFORM_COMMON_PY3_WHEEL_NAME=$(basename {{mlnx_platform_api_py3_wheel_path}})
```
Comment (jleveque) [inline files/build_templates/sonic_debian_extension.j2:648]: Nomenclature nit-pick: The vendor-implemented API package is simply called "sonic-platform", not "sonic-platform-common". The base API package is called "sonic-platform-common". I suggest you change "mlnx-sonic-platform-common" to "mlnx-sonic-platform" and "MLNX_PLATFORM_COMMON_PY3_WHEEL_NAME" to "MLNX_SONIC_PLATFORM_PY3_WHEEL_NAME". Also suggest doing the same for the Python 2 package above.

File: rules/docker-platform-monitor.mk [hunk 2]
```diff
@@ -23,6 +23,7 @@ $(DOCKER_PLATFORM_MONITOR)_PYTHON_WHEELS += $(SONIC_XCVRD)
 ifeq ($(PDDF_SUPPORT),y)
 $(DOCKER_PLATFORM_MONITOR)_PYTHON_WHEELS += $(PDDF_PLATFORM_API_BASE_PY2)
 endif
+$(DOCKER_PLATFORM_MONITOR)_PYTHON_WHEELS += $(SONIC_PLATFORM_API_PY3)
```
Comment (jleveque) [inline rules/docker-platform-monitor.mk:26]: Suggest moving this under line 16 to keep next to the Py2 version.

**Review by jleveque — CHANGES_REQUESTED** [review]

**Comment by jleveque** [comment]
@Junchao-Mellanox: Once this PR is merged, are we good to merge https://github.com/Azure/sonic-buildimage/pull/5886?

**Comment by Junchao-Mellanox** [comment]
> @Junchao-Mellanox: Once this PR is merged, are we good to merge #5886?

@keboliu I suppose you are working on the system health python3 migration, could you answer this comment?

File: platform/mellanox/mlnx-platform-api/sonic_platform/eeprom.py [hunk 3]
```diff
@@ -10,8 +10,10 @@
 import re
 
 if sys.version_info.major == 3:
+    import builtins as exceptions
```
Comment (jleveque) [inline platform/mellanox/mlnx-platform-api/sonic_platform/eeprom.py:13]: I believe you can remove this altogether. Please verify.

File: platform/mellanox/mlnx-platform-api/sonic_platform/eeprom.py [hunk 4]
```diff
@@ -10,8 +10,10 @@
...
     from io import StringIO
 else:
+    import exceptions
```
Comment (jleveque) [inline platform/mellanox/mlnx-platform-api/sonic_platform/eeprom.py:16]: I believe you can remove this altogether. Please verify.

**Review by jleveque — CHANGES_REQUESTED** [review]

**Review by jleveque — APPROVED** [review]

**Comment by keboliu** [comment]
> > @Junchao-Mellanox: Once this PR is merged, are we good to merge #5886?
> 
> @keboliu I suppose you are working on the system health python3 migration, could you answer this comment?

generally, system-health has a dependency on platform API python3 lib, and when it ready system-health should be able to run, however, I see that system-health itself still has some issue in terms of string handling. left comments in another PR https://github.com/Azure/sonic-buildimage/pull/5886

**Pull request merged**